python/ray/spark/utils.py

```diff
@@ -367,3 +369,43 @@ def get_spark_task_assigned_physical_gpus(gpu_addr_list):
         return [visible_cuda_dev_list[addr] for addr in gpu_addr_list]
     else:
         return gpu_addr_list
+
+
+def _port_acquisition_delay():
+    """
+    Use a file lock to delay the worker processes to ensure that port acquisition does not
+    create a resource contention issue due to a race condition
+
+    Returns: None
+    """
+
+    def acquire_lock(file):
+        mode = os.O_RDWR | os.O_CREAT | os.O_TRUNC
+        file_ref = os.open(file, mode)
+        locked_file_ref = None
+        lock_time = 10.0
+        start = current = time.time()
+        while current < start + lock_time:
+            try:
+                fcntl.flock(file_ref, fcntl.LOCK_EX | fcntl.LOCK_NB)
```

Comment:
> fcntl.LOCK_NB

What if we use blocking mode to acquire the lock ?

Reply:
Seemingly the logic of `_port_acquisition_delay` is not correct, your code logic is:

```
wait up to 10s to acquire the file lock
if acquiring lock succeeds, then release the lock immediately, if acquiring failed (timeout), then do nothing
```

But I expect the following logic:
```
wait until acquiring the lock successfully.
after acquiring the lock, holds the lock for 10s, and then release the lock.
```
Only this way, we can ensure we allocate 10s time for each task to run ray worker startup routine exclusively.